transcom/mymove
# INT B-20976 customer state management to avoid duplicate moves

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20976)

## Summary

We were running into an issue where customers were able to add an additional move after submitting their orders and then clicking the browser back button to go back to the add orders form, fill it out, and submit again. This was a huge no-no cause we really know all they wanted was just to update their orders information (we all make some oopsies sometimes).

At first we had this grand solution that was beautiful and wonderful and that was turned down, so we made it simpler.

Using Redux, we are tracking essentially where the customer is coming from to add the orders. We only want to let the customer add new orders if they are originating from the "Create a Move" button on the multi moves landing page. So if the customer clicks that and navigates to the add orders form, we let them in the house. If they try to access that route in any other capacity, we leave them outside with the dogs so they don't get mud on the carpets.

This PR does the following:
- update the `generalState` entity in state to add a `canAddOrders` value that defaults to `false`
- update that value to `true` when the customer clicks the `Create a Move` button
- update the `AddOrders.jsx` component to first check the `canAddOrders` value in state prior to loading and redirect the user to either the move home page or their default home page or let them add orders
- added and updated relevant tests

### How to test

1. Access the app as a customer
2. Create a new move and add your orders
3. Once you're on the upload attachments route, hit the browser back button
4. You should be redirected to the move home page that you just created
5. Try to access the `/orders/add` route from anywhere in the app
6. You should be redirected to the default home page
7. PLEASE TEST WITH BOTH MULTI MOVE FLAG ON AND OFF

## Screenshots
https://github.com/user-attachments/assets/5701f28a-d7a9-4244-b1ef-f8243cdbee33
